### ISSUE TITLE
pcap_EXPORTS definition missing from CMakeLists.txt

### ISSUE THREAD

**Issue by dmiller-nmap** [the-tcpdump-group/libpcap#810]
When building the DLL, the build will succeed but the DLL will have no exports because it is missing the `pcap_EXPORTS` macro definition checked for in `pcap/funcattrs.h`. The fix seems pretty simple, though I'm not very confident with CMake yet. Here's what worked for me (based on `libpcap-1.9.0` release tag):

```
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 905ba61a..97568e36 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1961,6 +1961,7 @@ if(WIN32)
         set_target_properties(${LIBRARY_NAME} PROPERTIES
             VERSION ${PACKAGE_VERSION_NOSUFFIX} # only MAJOR and MINOR are needed
         )
+       add_definitions(-Dpcap_EXPORTS)
     endif(BUILD_SHARED_LIBS)
     if(MSVC)
         # XXX For DLLs, the TARGET_PDB_FILE generator expression can be used to locate
```

**Comment by guyharris** [comment]
OK, it's *supposed* to be defined automatically by CMake, according to [the documentation for the `DEFINE_SYMBOL` property](https://cmake.org/cmake/help/latest/prop_tgt/DEFINE_SYMBOL.html):

> Define a symbol when compiling this target’s sources.
> 
> DEFINE_SYMBOL sets the name of the preprocessor symbol defined when compiling sources in a shared library. If not set here then it is set to target_EXPORTS by default (with some substitutions if the target is not a valid C identifier). This is useful for headers to know whether they are being included from inside their library or outside to properly setup dllexport/dllimport decorations.

but if you're overriding the setting of `LIBRARY_NAME` to make it something other than "pcap", it won't define `pcap_EXPORTS`, so we may want to explicitly set the `DEFINE_SYMBOL` property of the `${LIBRARY_NAME}` target to `pcap_EXPORTS`, so that it's `pcap_EXPORTS` regardless of whether it's building `pcap.dll` or `wpcap.dll` or `npcap.dll` or....

**Comment by dmiller-nmap** [comment]
oh, I see. Yes, I was overriding LIBRARY_NAME to be "wpcap" because that was by far the simplest way to integrate with Npcap. I will begin by also overriding DEFINE_SYMBOL instead of changing any libpcap code directly, but I'm glad to have raised the issue.

**Comment by guyharris** [comment]
Were you overriding `LIBRARY_NAME` with a `-D` option to CMake, or did you have to modify `CMakeLists.txt`?  If you had to modify `CMakeLists.txt`, that needs to be fixed.

**Comment by guyharris** [comment]
So does applying change 4d490932e08daefa778f58ea4694fc97486507ce to the 1.9.0 `CMakeLists.txt` work?

**Comment by infrastation** [comment]
Closing due to lack of feedback. The supposed fix is available in libpcap 1.9.1 and libpcap 1.10.0.
